src/operator/controllers/intents_reconcilers/database/database.go

```diff
@@ -39,51 +42,99 @@ func NewDatabaseReconciler(
 }
 
 func (r *DatabaseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	intents := &otterizev1alpha3.ClientIntents{}
+	clientIntents := &otterizev1alpha3.ClientIntents{}
 	logger := logrus.WithField("namespacedName", req.String())
-	err := r.client.Get(ctx, req.NamespacedName, intents)
+	err := r.client.Get(ctx, req.NamespacedName, clientIntents)
 	if err != nil && k8serrors.IsNotFound(err) {
-		logger.Info("No intents found")
+		logger.Info("No client intents found")
 		return ctrl.Result{}, nil
 	} else if err != nil {
 		return ctrl.Result{}, errors.Wrap(err)
 	}
 
-	if intents.Spec == nil {
+	if clientIntents.Spec == nil {
 		logger.Info("No specs found")
 		return ctrl.Result{}, nil
 	}
 
-	action := graphqlclient.DBPermissionChangeApply
-	if !intents.ObjectMeta.DeletionTimestamp.IsZero() {
-		action = graphqlclient.DBPermissionChangeDelete
+	action := otterizev1alpha3.DBPermissionChangeApply
+	if !clientIntents.ObjectMeta.DeletionTimestamp.IsZero() {
+		action = otterizev1alpha3.DBPermissionChangeDelete
 	}
 
-	var intentInputList []graphqlclient.IntentInput
-	for _, intent := range intents.GetCallsList() {
+	var dbIntents []otterizev1alpha3.Intent
+	for _, intent := range clientIntents.GetCallsList() {
 		if intent.Type != otterizev1alpha3.IntentTypeDatabase {
 			continue
 		}
-
-		intentInput := intent.ConvertToCloudFormat(intents.Namespace, intents.GetServiceName())
-		intentInputList = append(intentInputList, intentInput)
+		dbIntents = append(dbIntents, intent)
 	}
 
-	if len(intentInputList) == 0 {
+	if len(dbIntents) == 0 {
 		return ctrl.Result{}, nil
 	}
 
-	if err := r.otterizeClient.ApplyDatabaseIntent(ctx, intentInputList, action); err != nil {
-		errType, errMsg, ok := graphqlclient.GetGraphQLUserError(err)
-		if !ok || errType != graphqlclient.UserErrorTypeAppliedIntentsError {
-			r.RecordWarningEventf(intents, ReasonApplyingDatabaseIntentsFailed, "Failed applying database intents: %s", err.Error())
+	databaseToIntents, err := r.MapDBInstanceToIntents(dbIntents)
+	if err != nil {
+		r.RecordWarningEventf(clientIntents, ReasonApplyingDatabaseIntentsFailed, "Failed applying database clientIntents: %s", err.Error())
+		return ctrl.Result{}, errors.Wrap(err)
+	}
+
+	for databaseInstance, intents := range databaseToIntents {
```

Comment:
Same here. How do we handle database instances which had prior intents, and now, following the modified intent, have 0 intents? We should still remove any existing permissions.